parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -166,15 +167,19 @@ ParquetFileReader reader() {
   }
 
   private Map<Long, Long> generateOffsetToStartPos() {
-    ParquetFileReader fileReader = newReader(this.file, ParquetReadOptions.builder().build());
+
     Map<Long, Long> offsetToStartPos = new HashMap<>();
-    long curRowCount = 0;
-    for (int i = 0; i < fileReader.getRowGroups().size(); i += 1) {
-      BlockMetaData meta = fileReader.getRowGroups().get(i);
-      offsetToStartPos.put(meta.getStartingPos(), curRowCount);
-      curRowCount += meta.getRowCount();
-    }
 
+    try(ParquetFileReader fileReader = newReader(this.file, ParquetReadOptions.builder().build())) {
```

Comment:
Nit: missing space between `try` and `(`. This will probably fail validation.

Reply:
While you're changing this, can you remove `this.` from the variable reference? We use `this.` to distinguish setting instance fields, not getting instance fields.

Reply:
yup I will revert to code you posted above :)